src/commands/Minion/farm.ts

```diff
@@ -0,0 +1,313 @@
+			patchType,
+			userID: msg.author.id,
+			channelID: msg.channel.id,
+			quantity,
+			upgradeType,
+			planting: true,
+			duration,
+			msg,
+			type: Activity.Farming,
+			id: rand(1, 10_000_000),
+			finishDate: Date.now() + duration
+		};
+
+		// If user does not have something already planted, just plant the new seeds.
+		if (patchType.PatchStage === 0) {
+			const initialPatches = {
+				LastPlanted: plants.name,
+				PatchStage: 1,
+				PlantTime: currentDate,
+				LastQuantity: quantity,
+				LastUpgradeType: upgradeType,
+				LastPayment: payment,
+				IsHarvestable: true
+			};
+
+			msg.author.settings.update(getPatchType, initialPatches);
+
+			str += `${msg.author.minionName} is now planting ${quantity}x ${
+				plants.name
+			}. ${upgradeStr}${paymentStr}\nIt'll take around ${formatDuration(
+				duration
+			)} to finish. ${boostStr}`;
+		} else if (patchType.PatchStage === 1) {
+			const storeHarvestablePlant = patchType.LastPlanted;
+			const planted = Farming.Plants.find(
+				plants =>
+					stringMatches(plants.name, storeHarvestablePlant) ||
+					stringMatches(plants.name.split(' ')[0], storeHarvestablePlant)
+			);
+			if (!planted) {
+				throw `This error shouldn't happen. Just to clear possible undefined error`;
+			}
+
+			const lastPlantTime: number = patchType.PlantTime;
+			const difference = currentDate - lastPlantTime;
+			// initiate a cooldown feature for each of the seed types.
+			/* allows for a run of specific seed type to only be possible until the
+				previous run's plants has grown.*/
+			if (difference < planted.growthTime * Time.Minute) {
+				throw `Please come back when your crops have finished growing in ${formatDuration(
+					lastPlantTime + plants.growthTime * Time.Minute - currentDate
+				)}!`;
+			}
+
+			const storeHarvestableQuantity = patchType.LastQuantity;
+
+			if (planted.needsChopForHarvest === true) {
+				if (!planted.treeWoodcuttingLevel) return;
+				if (
+					currentWoodcuttingLevel < planted.treeWoodcuttingLevel &&
+					GP < 200 * storeHarvestableQuantity
+				) {
+					throw `Your minion remembers that they do not have ${planted.treeWoodcuttingLevel} woodcutting or the 200GP per patch required to be able to harvest the currently planted trees.`;
+				}
+			}
+
+			const updatePatches = {
+				LastPlanted: plants.name,
+				PatchStage: 1,
+				PlantTime: currentDate,
+				LastQuantity: quantity,
+				LastUpgradeType: upgradeType,
+				LastPayment: payment,
+				IsHarvestable: true
+			};
+
+			msg.author.settings.update(getPatchType, updatePatches);
+
+			str += `${
+				msg.author.minionName
+			} is now harvesting ${storeHarvestableQuantity}x ${storeHarvestablePlant}, and then planting ${quantity}x ${
+				plants.name
+			}. ${upgradeStr}${paymentStr}\nIt'll take around ${formatDuration(
+				duration
+			)} to finish. ${boostStr}`;
+		}
+
+		await addSubTaskToActivityTask(this.client, Tasks.SkillingTicker, data);
+		msg.author.incrementMinionDailyDuration(duration);
```

Comment:
remove this dailyDuration line, and instead put it in the task where the farming trip ends and they harvest stuff. anywhere is fine.